vms/types/blob_data.go

```diff
@@ -9,13 +9,33 @@ import (
 	"github.com/ava-labs/avalanchego/utils/formatting"
 )
 
+const nullStr = "null"
+
 // JSONByteSlice represents [[]byte] that is json marshalled to hex
 type JSONByteSlice []byte
 
 func (b JSONByteSlice) MarshalJSON() ([]byte, error) {
+	if b == nil {
+		return []byte(nullStr), nil
+	}
+
 	hexData, err := formatting.Encode(formatting.HexNC, b)
 	if err != nil {
 		return nil, err
 	}
 	return json.Marshal(hexData)
 }
+
+func (b *JSONByteSlice) UnmarshalJSON(jsonBytes []byte) error {
+	if string(jsonBytes) == nullStr {
+		return nil
```

Comment:
Should an explicit `null` empty `b` instead of being a no-op? This would be [in keeping](https://pkg.go.dev/encoding/json#Unmarshal:~:text=Unmarshal%20first%20handles%20the%20case%20of%20the%20JSON%20being%20the%20JSON%20literal%20null.%20In%20that%20case%2C%20Unmarshal%20sets%20the%20pointer%20to%20nil.) with `json.Unmarshal()`. It's also the inverse of the change you made to `JSONByteSlice.MarshalJSON()`.

Reply:
This implementation is keeping with the best-practices of `UnmarshalJSON` afaik.

> By convention, to approximate the behavior of [Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal) itself, Unmarshalers implement UnmarshalJSON([]byte("null")) as a no-op.

Reply:
Looks like that comment on `Unmarshal()` is incorrect (or I misunderstood it). [Confirmed in playground](https://go.dev/play/p/Szzlkzrkerv).